### PR TITLE
Custom colors via config.json

### PR DESCRIPTION
**Pull request by alkama**
The expected input format for the color is "RRGGBBAA"

Note:
- uses `std::stoul()` to convert hex as string to unsigned long. No error is handled for the conversion so a bad format there will result in crash :D PEBKAC
- color struct are not memcopied but relies instead on struct copy via assignment to be handled correctly (I think it should)

### CODE REVIEW

**Comment by alkama** [comment]
Wokay, apparently the linux build is not targetting c++11 minimum, so it lacks std::stoul...
Will probably have to change the CMakeLists.txt to force c++11 standard

**Comment by alkama** [comment]
Gcc forced to support standard c++11 for the bonzomatic executable compilation

**Comment by alkama** [comment]
Finally, the expected input format for the color is "AARRGGBB" or "RRGGBB" in hexadecimal.

Minimal c++ standard required is c++11 so that:
- we now can use auto (will clean the json loading a lot)
- color parsing uses `std::stoul()` to convert hex as string to unsigned long
- structs can be initialized "on site"
- structs can be copy assigned without requiring memcopy

**Comment by alkama** [comment]
Now we also handle background color for characters (in case people want to set opacity to 0 and have background only behind text.

Text selection background color is overridable too so that it doesn't depend on opacity.

I also removed the splitter between line numbers and code forcefully (it's ugly but it works).

File: external/scintilla/src/EditView.cxx [hunk 1]
```diff
@@ -880,7 +880,11 @@ void EditView::DrawEOL(Surface *surface, const EditModel &model, const ViewStyle
 		if (background.isSet) {
 			surface->FillRectangle(rcSegment, background);
 		} else if (line < model.pdoc->LinesTotal() - 1) {
+			// Note(Bonzomatic): ULGY HACK TO REMOVE CHAR BACKGROUND ON END OF LINES
```
Comment (Gargaj) [inline external/scintilla/src/EditView.cxx]: "ULGY"? :)

Comment (alkama) [reply external/scintilla/src/EditView.cxx]: Hahaha omg, it's even more accurate like that :D

**Comment by alkama** [comment]
Bad typo was bad :D fixed that hehehe

**Comment by alkama** [comment]
Fixed a problem in case of global opacity set to fully transparent:
It previously stopped colorizing the content but also meant the text is not lexed anymore. When fully transparent gui is used this text is not even adopting the 'charBackground' theme color. By keeping on lexing and colorizing the #if #else #endif content, the problem do not exist anymore.

Also adopting the @blackle config setting to lex/colorize the preprocessor directives.

**Pull request merged**